### PR TITLE
[TTA-2250] fei goals query for JDI

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change
Query added to /src/queries dir for reuse and integration into future JDI (Jorge Data Interface)

## How to test

- Run query
- Run query using any combination of the JDI flags

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2250

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Comment by thewatermethod** [comment]
can you add this directory to the .cfignore for now?

File: src/queries/fei-goals.sql [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			gr.number "grant number",
+			gr."regionId",
+			b.id "goalId",
+			b."goalTemplateId",
+			b.name "goal text",
+			b."createdAt",
+			b.status,
+			b."createdVia",
+			b."onApprovedAR",
+			gfr.response
```
Comment (thewatermethod) [inline src/queries/fei-goals.sql:54]: I take it that historically it isn't a problem that this column is "Response" when the ticket is asking for "Root cause"

Comment (hardwarehuman) [reply src/queries/fei-goals.sql:54]: "response" is the general term for a response to a question prompt associated with a goal template. Currently we only have the FEI question about root causes, but we could have other questions in the future

Comment (thewatermethod) [reply src/queries/fei-goals.sql:54]: I understand that, more asking about the expectations of the people using this CSV

Comment (thewatermethod) [reply src/queries/fei-goals.sql:54]: But yeah, if we've been using this SQL to generate a CSV and no one has cared, I'm sure it's fine

File: src/queries/fei-goals.sql [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/**
+* This query collects all the FEI and near FEI goals based on several curiteriea.
```
Comment (AdamAdHocTeam) [inline src/queries/fei-goals.sql:2]: criteria?

File: src/queries/fei-goals.sql [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		JOIN "Grants" gr
+		ON b."grantId" = gr.id
+		JOIN "Recipients" r
+		ON gr."recipientId" = r.id
+		WHERE (NULLIF(current_setting('jdi.regionIds', true), '') IS NULL
+			   OR gr."regionId" in (
+				   SELECT value::integer AS my_array
+			       FROM json_array_elements_text(COALESCE(NULLIF(current_setting('jdi.regionIds', true), ''),'[]')::json) AS value
+			   ))
+		AND (NULLIF(current_setting('jdi.recipients', true), '') IS NULL
```
Comment (AdamAdHocTeam) [inline src/queries/fei-goals.sql:67]: Why do we use current_setting?

File: src/queries/fei-goals.sql [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+			   OR r.name in (
+				   SELECT value::text AS my_array
+			       FROM json_array_elements_text(COALESCE(NULLIF(current_setting('jdi.recipients', true), ''),'[]')::json) AS value
```
Comment (AdamAdHocTeam) [inline src/queries/fei-goals.sql:70]: json_array_elements_text vs UNNEST is there a diff?

File: src/queries/fei-goals.sql [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+			   OR b.name in (
+				   SELECT value::text AS my_array
+			       FROM json_array_elements_text(COALESCE(NULLIF(current_setting('jdi.goal', true), ''),'[]')::json) AS value
+			   ))
+		AND (NULLIF(current_setting('jdi.status', true), '') IS NULL
+			   OR b.status in (
+				   SELECT value::text AS my_array
+			       FROM json_array_elements_text(COALESCE(NULLIF(current_setting('jdi.status', true), ''),'[]')::json) AS value
+			   ))
+		AND (NULLIF(current_setting('jdi.createdVia', true), '') IS NULL
```
Comment (AdamAdHocTeam) [inline src/queries/fei-goals.sql:87]: Each one of the blocks seems to follow the same pattern for different fields can you put a comment at the top just explaining what this is doing?

File: src/queries/fei-goals.sql (hunk 3 above)
Comment (GarrettEHill) [reply src/queries/fei-goals.sql:67]: This is a way to access variables within Postgres. This is the same method we use for setting/retrieving data for the audit log.

File: src/queries/fei-goals.sql (hunk 4 above)
Comment (GarrettEHill) [reply src/queries/fei-goals.sql:70]: json_array_elements_text - data coming in as json
unnest - data as a Postgres array

**Pull request merged**